### PR TITLE
Simplifying header and footer to use data

### PR DESCRIPTION
**Pull request by bapplejax**
## Description
This work refactors the header and footer components to utilize data instead of hard-coded links.  This code also adds a helper utility function to format strings and a corresponding test.

## Related Issue
https://metrostarsys.sharepoint.com/sites/FrontendDevCoP/Lists/Tasks/DispForm.aspx?ID=31&e=S7b57s

## Motivation and Context
The idea of this work is to allow users to define their own data and have links automatically generated.

## How Has This Been Tested?
A test was created for the helper function and all current testing passed.

### CODE REVIEW

File: src/utils/helpers.ts [hunk 1]
```diff
@@ -0,0 +1,6 @@
+// Utility function to format string
+export const lowercaseHyphenateString = function (dirtyString: string) {
```
Comment (jbouder) [inline src/utils/helpers.ts:2]: Lets make this a arrow function for consistency. Should be something like `export const lowercaseHyphenateString = (dirtyString: string) => {`

File: src/utils/helpers.test.ts [hunk 2]
```diff
@@ -0,0 +1,14 @@
+import { lowercaseHyphenateString } from './helpers';
+
+describe('Generic Helper Functions', () => {
+  const OLD_ENV = process.env;
+  beforeEach(() => {
+    process.env = { ...OLD_ENV };
+  });
```
Comment (jbouder) [inline src/utils/helpers.test.ts:7]: These can be removed as they are not needed or these tests

**Review by jbouder — CHANGES_REQUESTED** [review]
Looks great and works! Just a few minor updates please.

File: src/components/footer/footer.tsx [hunk 3]
```diff
@@ -13,6 +13,49 @@ export const Footer = (): React.ReactElement => {
     window.scrollTo(0, 0);
   };
 
+  const FOOTER_LINKS = [
```
Comment (jbouder) [inline src/components/footer/footer.tsx:16]: Let’s put this in the constants.ts file

File: src/components/footer/footer.tsx [hunk 4]
```diff
@@ -13,6 +13,49 @@ export const Footer = (): React.ReactElement => {
...
+      name: 'Contact Us',
+      url: '/contact-us',
+    },
+    {
+      name: 'About',
+      url: '/about',
+    },
+  ];
+
+  const SOCIAL_LINKS = [
```
Comment (jbouder) [inline src/components/footer/footer.tsx:31]: Same, put in constants

File: src/components/header/header.tsx [hunk 5]
```diff
@@ -18,6 +19,26 @@ export const Header = (): React.ReactElement => {
     setShowMenu(!showMenu);
   };
 
+  // Header link data for signed in users
+  const HEADER_LINKS_SIGNED_IN = [
```
Comment (jbouder) [inline src/components/header/header.tsx:23]: Same, constants file

File: src/components/header/header.tsx [hunk 6]
```diff
@@ -18,6 +19,26 @@ export const Header = (): React.ReactElement => {
...
+      url: '/',
+    },
+    {
+      name: 'Dashboard',
+      url: '/dashboard',
+    },
+  ];
+
+  // Header link data for anonymous users
+  const HEADER_LINKS_SIGNED_OUT = [
```
Comment (jbouder) [inline src/components/header/header.tsx:35]: Same, constants file

File: src/utils/helpers.ts (hunk 1 above)
Comment (bapplejax) [reply src/utils/helpers.ts]: gotcha

**Pull request merged**